server/data/data-cassandra/src/main/java/org/apache/james/user/cassandra/CassandraUsersDAO.java

```diff
@@ -78,7 +85,20 @@ public CassandraUsersDAO(Session session) {
             .value(REALNAME, bindMarker(REALNAME))
             .value(PASSWORD, bindMarker(PASSWORD))
             .value(ALGORITHM, bindMarker(ALGORITHM))
+            .value(HASHINGMODE, bindMarker(HASHINGMODE))
             .ifNotExists());
+        this.preferredAlgorithm = Algorithm.of("SHA-512", "salted");
+        this.fallbackAlgorithm = Algorithm.of("SHA1", "plain");
+    }
+
+    @Override
+    public void configure(HierarchicalConfiguration<ImmutableNode> config) throws ConfigurationException {
+        preferredAlgorithm = Algorithm.of(
+            config.getString("algorithm", "SHA-512"),
+            config.getString("hashingMode", "salted"));
```

Comment:
`HashingMode.SALTED.asString()` is better?

Reply:
Good idea, fixed in latest push.